src/geode/mesh/core/solid_mesh.cpp

```diff
@@ -0,0 +1,1093 @@
+        }
+
+        void update_edge_vertex( std::array< index_t, 2 > edge_vertices,
+            const index_t edge_vertex_id,
+            const index_t new_vertex_id )
+        {
+            auto updated_edge_vertices = edge_vertices;
+            updated_edge_vertices[edge_vertex_id] = new_vertex_id;
+            if( edge_vertices[0] < edge_vertices[1] )
+            {
+                Edges::add_facet( std::move( updated_edge_vertices ) );
+                Edges::remove_facet( std::move( edge_vertices ) );
+            }
+        }
+
+        void update_facet_vertices( absl::Span< const index_t > old2new )
+        {
+            Facets::update_facet_vertices( old2new );
+        }
+
+        void update_edge_vertices( absl::Span< const index_t > old2new )
+        {
+            Edges::update_facet_vertices( old2new );
+        }
+
+        void remove_facet( PolyhedronFacetVertices facet_vertices )
+        {
+            Facets::remove_facet( std::move( facet_vertices ) );
+        }
+
+        void remove_edge( std::array< index_t, 2 > edge_vertices )
+        {
+            if( edge_vertices[0] < edge_vertices[1] )
+            {
+                Edges::remove_facet( std::move( edge_vertices ) );
+            }
+        }
+
+        std::vector< index_t > delete_facets(
+            const std::vector< bool >& to_delete )
+        {
+            return Facets::delete_facets( to_delete );
+        }
+
+        std::vector< index_t > delete_edges(
+            const std::vector< bool >& to_delete )
+        {
+            return Edges::delete_facets( to_delete );
+        }
+
+        std::vector< index_t > remove_isolated_facets()
+        {
+            return Facets::clean_facets();
+        }
+
+        std::vector< index_t > remove_isolated_edges()
+        {
+            return Edges::clean_facets();
+        }
+
+        bool isolated_vertex( index_t vertex_id ) const
+        {
+            return polyhedron_around_vertex( vertex_id ) == PolyhedronVertex{};
+        }
+
+        bool isolated_edge( index_t edge_id ) const
+        {
+            return Edges::get_counter( edge_id ) == 0;
+        }
+
+        bool isolated_facet( index_t facet_id ) const
+        {
+            return Facets::get_counter( facet_id ) == 0;
+        }
+
+        AttributeManager& polyhedron_attribute_manager() const
+        {
+            return polyhedron_attribute_manager_;
+        }
+
+        AttributeManager& facet_attribute_manager() const
+        {
+            return Facets::facet_attribute_manager();
+        }
+
+        AttributeManager& edge_attribute_manager() const
+        {
+            return Edges::facet_attribute_manager();
+        }
+
+        void overwrite_facets( const Facets& from )
+        {
+            Facets::overwrite( from );
+        }
+
+        void overwrite_edges( const Edges& from )
+        {
+            Edges::overwrite( from );
+        }
+
+    private:
+        void convert_attribute_to_abseil()

```

Comment:
should allow downward compatiblity? really cool

Reply:
that's not new. It is an old convertion